superset-frontend/src/explore/components/controls/FilterControl/AdhocFilter/index.js

```diff
@@ -135,6 +135,10 @@ export default class AdhocFilter {
 
   getDefaultLabel() {
     const label = this.translateToSql();
+    // If returned value is an object after changing dataset
+    if (typeof label === 'object') {
+      return label.column_name ?? '';
```

Comment:
Do I understand correctly that the bug occurs when `adhocFIlter.subject` is an object and not a string? If so, maybe we could move the logic of extracting column_name to `getSimpleSQLExpression`, line 298 - `let expression = subject ?? '';`?